piptools/scripts/sync.py

```diff
@@ -147,3 +147,68 @@ def cli(
             ask=ask,
         )
     )
+
+
+def _build_install_flags(
+    finder,
+    no_index=False,
+    index_url=None,
+    extra_index_url=None,
+    trusted_host=None,
+    find_links=None,
+    user_only=False,
+    cert=None,
+    client_cert=None,
+):
+    """
+    Builds install flags with the given finder and CLI options.
+    """
+    result = []
+
+    # Build --index-url/--extra-index-url/--no-index
+    if no_index:
+        result.append("--no-index")
+    elif index_url:
+        result.extend(["--index-url", index_url])
+    elif finder.index_urls:
+        finder_index_url = finder.index_urls[0]
+        if finder_index_url != PyPIRepository.DEFAULT_INDEX_URL:
+            result.extend(["--index-url", finder_index_url])
+        for extra_index in finder.index_urls[1:]:
+            result.extend(["--extra-index-url", extra_index])
+    else:
+        result.append("--no-index")
+
+    for extra_index in extra_index_url or []:
+        result.extend(["--extra-index-url", extra_index])
+
+    # Build --trusted-hosts
+    if PIP_VERSION < (19, 2):
+        finder_trusted_hosts = (host for _, host, _ in finder.secure_origins)
+    else:
+        finder_trusted_hosts = finder.trusted_hosts
+    for host in itertools.chain(trusted_host or [], finder_trusted_hosts):
+        result.extend(["--trusted-host", host])
+
+    # Build --find-links
+    for link in itertools.chain(find_links or [], finder.find_links):
+        result.extend(["--find-links", link])
+
+    # Build format controls --no-binary/--only-binary
+    for format_control in ("no_binary", "only_binary"):
+        formats = getattr(finder.format_control, format_control)
+        if formats:
```

Comment:
_Pro tip:_ You can reduce the nesting by reversing checks and interrupting the loop — then you can have the whole block dedented.
```suggestion
        if not formats:
            continue
```

Reply:
Indeed. Thanks!